txpipe/data_types/types.py

```diff
@@ -12,16 +12,36 @@ def metacalibration_names(names):
         out += [name + '_' + s for s in suffices]
     return out
 
-class MetacalCatalog(HDFFile):
+class ShearCatalog(HDFFile):
     """
-    A metacal output catalog
+    A generic shear catalog 
     """
     # These are columns
-    required_datasets = ['metacal/mcal_g1', 'metacal/mcal_g1_1p', 
-        'metacal/mcal_g2', 'metacal/mcal_flags', 'metacal/ra',
-        'metacal/mcal_T']
+    
+    def read_catalog_info(self):
+        try:
+            group = self.file['shear']
+            info = dict(group.attrs)
+        except:
+            raise ValueError(f"Unable to read shear catalog")
+        shear_catalog_type = info.get('catalog_type')
+        return shear_catalog_type
 
-    # Add methods for handling here ...
+    def validate(self):
+        
+        shear_catalog_type = self.read_catalog_info()
+        
+        if shear_catalog_type=='metacal':
```

Comment:
The required datasets here are not being used.  I think you can just delete - the validation stuff has never been as useful as I'd thought.